### PR TITLE
feat: add error message to async components

### PR DESCRIPTION
**Pull request by un-versed**
Solves https://github.com/denoland/fresh/issues/405

image

### CODE REVIEW

File: src/server/context.ts [hunk 1]
```diff
@@ -400,6 +400,11 @@ export class ServerContext {
           if (route.component === undefined) {
             throw new Error("This page does not have a component to render.");
           }
+
+          if (typeof route.component === 'function' && route.component.constructor.name === "AsyncFunction") {
+            throw new Error("Async components are not supported.");
```
Comment (lucacasonato) [inline src/server/context.ts]: Can you add a link to the data fetching docs here too?

Comment (un-versed) [reply src/server/context.ts]: Done

File: src/server/context.ts [hunk 2]
```diff
@@ -400,6 +400,11 @@ export class ServerContext {
           if (route.component === undefined) {
             throw new Error("This page does not have a component to render.");
           }
+
+          if (typeof route.component === 'function' && route.component.constructor.name === "AsyncFunction") {
+            throw new Error("Async components are not supported. Fetch data inside of the loader, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data");
```
Comment (lucacasonato) [inline src/server/context.ts]:
```suggestion
            throw new Error("Async components are not supported. Fetch data inside of a route handler, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data");
```

File: src/server/context.ts [hunk 3]
```diff
@@ -400,6 +400,16 @@ export class ServerContext {
           if (route.component === undefined) {
             throw new Error("This page does not have a component to render.");
           }
+
+          if (
+            typeof route.component === "function" &&
+            route.component.constructor.name === "AsyncFunction"
+          ) {
+            throw new Error(
+              "Async components are not supported. Fetch data inside of the loader, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data",
```
Comment (un-versed) [inline src/server/context.ts]: https://github.com/denoland/fresh/pull/413#discussion_r913696364

```suggestion
              "Async components are not supported. Fetch data inside of a route handler, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data",
```

**Review by lucacasonato — APPROVED** [review]
Thank you!

**Pull request merged**